pkg/network/tracer/tracer.go

```diff
@@ -865,6 +876,70 @@ func (t *Tracer) conntrackExists(ctr *cachedConntrack, conn *ConnTuple) bool {
 	return ok
 }
 
+func (t *Tracer) connVia(cs *network.ConnectionStats) {
+	r, ok := t.routeCache.Get(cs.Source, cs.Dest, cs.NetNS)
+	if !ok {
+		return
+	}
+
+	// if there is no gateway, we don't need to add subnet info
+	// for gateway resolution in the backend
+	if util.NetIPFromAddress(r.Gw).IsUnspecified() {
+		return
+	}
+
+	s, ok := t.subnetCache[r.IfIndex]
```

Comment:
Just to verify -- this is a non-namespaced IfIndex ?

Reply:
Yes, we only care about the root namespace here since that is going to have the route for egress from the host.